src/lib/expression/abstract_predicate_expression.cpp

```diff
@@ -0,0 +1,44 @@
+#include "abstract_predicate_expression.hpp"
+
+#include <sstream>
+
+#include "boost/functional/hash.hpp"
+
+namespace opossum {
+
+AbstractPredicateExpression::AbstractPredicateExpression(
+    const PredicateCondition predicate_condition, const std::vector<std::shared_ptr<AbstractExpression>>& arguments)
+    : AbstractExpression(ExpressionType::Predicate, arguments), predicate_condition(predicate_condition) {}
+
+DataType AbstractPredicateExpression::data_type() const {
+  // Should be Bool, but we don't have that.
+  return DataType::Int;
```

Comment:
Worth a thought?

Reply:
having another member in DataType would be more compilation time/binary size. Let's evaluate that later. (added to list of things to discuss)